ads-service-fixed/Dockerfile

```diff
@@ -1,6 +1,20 @@
-FROM python:3.9.1-alpine3.12
-RUN apk add build-base eudev-dev cython postgresql-dev linux-headers
-COPY requirements.txt /app/requirements.txt
+# syntax = docker/dockerfile:1.2
+# ^ This enables the new BuildKit stable syntax which can be
+# run with the DOCKER_BUILDKIT=1 environment variable in your
+# docker build command (see build.sh)
+FROM python:3.9.1-slim-buster
+
+# Update, upgrade, and cleanup debian packages
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update && \
+    apt-get upgrade --yes && \
+    apt-get install --yes build-essential libpq-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Bring in app
 WORKDIR /app
-ADD . /app
-RUN pip install -r requirements.txt
+COPY . .
+
+# Install dependencies via pip and avoid caching build artifacts
+RUN pip install --no-cache-dir -r requirements.txt
```

Comment:
You can optimize further by removing the Random-Words dependency. It's not used in ads.py (though it is imported and assigned to a variable). Furthermore, it now requires having an API key for some random words service. Discounts uses it too, but also has a words file that it uses in bootstrap. That's a superior solution, should the POST endpoints need to work in the future.

Reply:
Good catch there! I'll see about doing this in a separate pass from this one since it'll affect both services.

Reply:
Made #68 to track that